lab-scott/lib/storage.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+const debug = require('debug')('http:storage');
+const storage = {};
+
+module.exports = exports = {};
+
+//POST
+exports.createItem = function(schema, item) {
+  debug('#createItem');
+
+  if(!schema) return Promise.reject(new Error('schema required'));
+  if(!item) return Promise.reject(new Error ('item required'));
+  if(!storage[schema]) storage[schema] = {};
+
+  storage[schema][item.id] = item;
+
+  return Promise.resolve(item);
+};
+
+//GET
+exports.fetchItem = function(schema, id) {
+  debug('#fetchItem');
+
+  return new Promise((resolve, reject) => {
+    if(!schema) return reject(new Error('schema required'));
+    if(!id) return reject(new Error('id required'));
+
+    let schemaName = storage[schema];
+    if(!schemaName) return reject(new Error('schema not found'));
+
+    let item = schemaName[id];
+    if(!item) return reject(new Error('item not found'));
+
+    resolve(item);
+  });
+};
+
+
+//DELETE
+exports.deleteItem = function(schema, id){
+  debug('#fetchItem');
+
+  return new Promise((reslove, reject) => {
+    if(!schema) return Promise.reject(new Error('schema required'));
+    if(!id) return reject(new Error('id required'));
+
+    let schemaName = storage[schema];
+    if(!schemaName) return reject(new Error('schema not found'));
+
+    let item = schemaName[id];
+    if(!item) return reject(new Error('item not found'));
+
+    delete storage[schema];
+    reslove();
+  });
+};
+
+/////////////////////////////////////////////////////////////////////
+
+
+// const storage = {}
```

Comment:
Not going to dock you points for this, but in the future try not to put huge blocks of commented code in your commits. You can use `git add -p <file>` to add part of a file if you don't want to remove it for some reason.